### PR TITLE
Assert in pthread_self that the pointer is valid

### PR DESCRIPTION
**Pull request by kripken**
#8955 may cause issues with static constructors using `pthread_self`. Until we fix that, at least show an assertion that may help people diagnose issues.

(note we can't show a nice string as this code ends up inside asm.js in fastcomp; but the comment and the code should still help point in the right direction)

### CODE REVIEW

File: src/library_pthread.js [hunk 1]
```diff
@@ -25,7 +25,6 @@ var LibraryPThread = {
       __register_pthread_ptr(PThread.mainThreadBlock, /*isMainBrowserThread=*/!ENVIRONMENT_IS_WORKER, /*isMainRuntimeThread=*/1);
       _emscripten_register_main_browser_thread_id(PThread.mainThreadBlock);
       // Initialize data on main pthread now that we have initialized the thread id
```
Comment (tlively) [inline src/library_pthread.js]: I believe this comment is now stale.

File: system/lib/libc/musl/src/misc/emscripten_pthread.c [hunk 2]
```diff
@@ -10,13 +10,16 @@ pthread_t pthread_self(void) {
...
 
 #if __EMSCRIPTEN_PTHREADS__
-// Needs to be called after PThread.initRuntime, which in turn needs to be
-// called after constructors have run and memory is initialized.
+EM_JS(void, initPthreadsJS, (void), {
+  PThread.initRuntime();
+})
+#endif
+
 EMSCRIPTEN_KEEPALIVE
```
Comment (tlively) [inline system/lib/libc/musl/src/misc/emscripten_pthread.c]: This EMSCRIPTEN_KEEPALIVE used to be conditional but now it is not. Is that intentional?

File: src/library_pthread.js (hunk 1 above)
Comment (kripken) [reply src/library_pthread.js]: Indeed, fixing!

File: system/lib/libc/musl/src/misc/emscripten_pthread.c (hunk 2 above)
Comment (kripken) [reply system/lib/libc/musl/src/misc/emscripten_pthread.c]: Not intentional, yeah, and I see it broke a metadce test, good we kept those around ;)

**Comment by kripken** [comment]
Good to know about beta, thanks @tlively - meanwhile I've updated to nightly though, hopefully will save more time in the future to skip ahead (or maybe more noise, who knows...).

**Comment by dschuff** [comment]
This seems to have broken the metadce tests on the release builder: https://ci.chromium.org/p/emscripten-releases/builders/try/linux/b8907879386908703712
Do we need a test update or is there an LLVM change that has to be rolled together?

**Pull request merged**